boot/bootutil/src/swap_move.c

```diff
@@ -234,6 +234,28 @@ boot_status_internal_off(const struct boot_status *bs, int elem_sz)
     return off;
 }
 
+static int app_max_sectors(struct boot_loader_state *state)
+{
+    uint32_t sz = 0;
+    uint32_t sector_sz;
+    uint32_t trailer_sz;
+    uint32_t first_trailer_idx;
+
+    sector_sz = boot_img_sector_size(state, BOOT_PRIMARY_SLOT, 0);
+    trailer_sz = boot_trailer_sz(BOOT_WRITE_SZ(state));
+    first_trailer_idx = boot_img_num_sectors(state, BOOT_PRIMARY_SLOT) - 1;
+
+    while (1) {
+        sz += sector_sz;
+        if  (sz >= trailer_sz) {
+            break;
+        }
+        first_trailer_idx--;
+    }
```

Comment:
This is size or division optimization for
```
first_trailer_idx = boot_img_num_sectors(state, BOOT_PRIMARY_SLOT) -  (trailer_sz + sector_sz - 1) / sector_sz;
```
?
I know that this was here previously, just wondering why.

Reply:
Was added 5 years ago like that apparently, could be changed to the summarised form if wanted

Reply:
Nah. Lets fix it first, than we can go through these things and do some refactoring.